.azure-pipelines/ultimate-pipeline.yml

```diff
@@ -876,6 +880,201 @@ stages:
     - script: tracer\build.cmd CheckBuildLogsForErrors
       displayName: Check logs for errors
 
+- stage: asan_tests_windows_profiler
+  condition: > 
+    and(
+      succeeded(), 
+      eq(variables['isScheduledBuild'], 'False'),
+      eq(dependencies.generate_variables.outputs['generate_variables_job.generate_variables_step.IsProfilerChanged'], 'True')
```

Comment:
🤔 I don't want to run tests for nothing (especially for avoiding more noise with potentially more flaky tests), but as we do release one package, shouldn't we run all tests all the time? Could we miss cases where changes on tracer side could break the profiler? (plus anyway, I think this variable ins't large enough, see my comment in `tracer/build/_build/Build.VariableGenerations.cs`

If we decide not to, in that case, shouldn't we not run the tracer tests when there are only profiler changes ?

Reply:
I agree with you, we don't want to run tests for nothing, but on the other hand, we are struggling with excessive CI build times, so it's a question of what trade-offs we want and where do we draw the line. 

As for missing cases, the profiler has a relatively small dependency on the tracer (the Code Hotspots / trace-id and span-id retrieval), and they don't share code other than the native loader. If native loader changed, all tests should run, no exceptions (this is exactly the reason I've introduced `isNativeLoaderChanged` variable in this PR. It will force all other `changed` variables to be true.)

Another option is to run "nightly" builds which will run all tests for all products.

> If we decide not to, in that case, shouldn't we not run the tracer tests when there are only profiler changes ?

I think this is really good idea, we already have `isTracerChanged` variable, we just need to start use it in test steps. 
This could save us a lot of CI build time.

Reply:
> Another option is to run "nightly" builds which will run all tests for all products.

Ok, this is what we went for for now. cf #2893. We had to do it now because we realized that our current release process is broken and this fixed it.

Reply:
You'll just have to rename 'isScheduledBuild' to 'isBenchmarksOnlyBuild' in your PR as well